modules/build/src/main/scala/scala/build/bsp/BspReloadableOptions.scala

```diff
@@ -0,0 +1,26 @@
+package scala.build.bsp
+
+import java.util.concurrent.atomic.AtomicReference
+
+import scala.build.Logger
+import scala.build.blooprifle.BloopRifleConfig
+import scala.build.options.BuildOptions
+
+case class BspReloadableOptions(
+  buildOptions: BuildOptions,
+  bloopRifleConfig: BloopRifleConfig,
+  logger: Logger,
+  verbosity: Int
+)
+
+object BspReloadableOptions {
+  case class Reference(getReloaded: () => BspReloadableOptions) {
```

Comment:
This shouldn't be a case class, as because of the `ref` field, `ref` and `ref.copy()` are not equivalent while case class makes them equal.

Reply:
fixed